src/json.ts

```diff
@@ -490,130 +509,193 @@ export function getJsonRpcIdValidator(options?: JsonRpcValidatorOptions) {
 export function validateJsonAndGetSize(
   jsObject: unknown,
   skipSizingProcess = false,
-): [isValid: boolean, plainTextSizeInBytes: number] {
+): ValidatedJson {
```

Comment:
Is there any way we can improve the variable names? Not sure what could be better, but semantically it feels weird that this function can return a value with a type that is "ValidatedJson" but can actually be invalid Json.